TestDashboardRedirect is unstable

## Bug Report

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What did you do?

run ci

### What did you expect to see?

### What did you see instead?
```
[2020-11-17T07:42:58.325Z] ----------------------------------------------------------------------

[2020-11-17T07:42:58.325Z] FAIL: service_test.go:76: serverTestSuite.TestDashboardRedirect

[2020-11-17T07:42:58.325Z] 

[2020-11-17T07:42:58.325Z] service_test.go:77:

[2020-11-17T07:42:58.325Z]     s.testDashboard(c, false)

[2020-11-17T07:42:58.325Z] service_test.go:90:

[2020-11-17T07:42:58.325Z]     c.Assert(resp.StatusCode, Equals, code)

[2020-11-17T07:42:58.325Z] ... obtained int = 307

[2020-11-17T07:42:58.325Z] ... expected int = 200

[2020-11-17T07:42:58.325Z] 

[2020-11-17T07:42:58.325Z] OOPS: 2 passed, 1 FAILED

[2020-11-17T07:42:58.325Z] --- FAIL: Test (30.67s)
```
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/pd_test/detail/pd_test/13859/pipeline

### What version of PD are you using (`pd-server -V`)?

```
2023-09-13T04:14:38.6451997Z --- FAIL: TestDashboardTestSuite (28.37s)
2023-09-13T04:14:38.6452385Z     --- FAIL: TestDashboardTestSuite/TestDashboardRedirect (9.92s)
2023-09-13T04:14:38.6452620Z         service_test.go:85: 
2023-09-13T04:14:38.6453346Z             	Error Trace:	/home/runner/work/pd/pd/tests/dashboard/service_test.go:85
2023-09-13T04:14:38.6454575Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:100
2023-09-13T04:14:38.6455479Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:154
2023-09-13T04:14:38.6456358Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:72
2023-09-13T04:14:38.6456608Z             	Error:      	Not equal: 
2023-09-13T04:14:38.6456934Z             	            	expected: 200
2023-09-13T04:14:38.6457445Z             	            	actual  : 307
2023-09-13T04:14:38.6457903Z             	Test:       	TestDashboardTestSuite/TestDashboardRedirect
2023-09-13T04:14:38.6458129Z         service_test.go:85: 
2023-09-13T04:14:38.6458879Z             	Error Trace:	/home/runner/work/pd/pd/tests/dashboard/service_test.go:85
2023-09-13T04:14:38.6459846Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:101
2023-09-13T04:14:38.6460812Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:154
2023-09-13T04:14:38.6461759Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:72
2023-09-13T04:14:38.6462027Z             	Error:      	Not equal: 
2023-09-13T04:14:38.6462372Z             	            	expected: 401
2023-09-13T04:14:38.6462868Z             	            	actual  : 307
2023-09-13T04:14:38.6463464Z             	Test:       	TestDashboardTestSuite/TestDashboardRedirect
```
https://github.com/tikv/pd/actions/runs/6167864066/job/16739488684

maybe related ci failed `TestDashboardProxy`

```
2023-09-13T09:39:00.1844923Z --- FAIL: TestDashboardTestSuite (28.04s)
2023-09-13T09:39:00.1845337Z     --- FAIL: TestDashboardTestSuite/TestDashboardProxy (12.87s)
2023-09-13T09:39:00.1845599Z         service_test.go:85: 
2023-09-13T09:39:00.1846236Z             	Error Trace:	/home/runner/work/pd/pd/tests/dashboard/service_test.go:85
2023-09-13T09:39:00.1847219Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:101
2023-09-13T09:39:00.1848185Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:154
2023-09-13T09:39:00.1849153Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:76
2023-09-13T09:39:00.1849421Z             	Error:      	Not equal: 
2023-09-13T09:39:00.1849766Z             	            	expected: 401
2023-09-13T09:39:00.1850106Z             	            	actual  : 508
2023-09-13T09:39:00.1850584Z             	Test:       	TestDashboardTestSuite/TestDashboardProxy
2023-09-13T09:39:00.1850824Z         service_test.go:85: 
2023-09-13T09:39:00.1851443Z             	Error Trace:	/home/runner/work/pd/pd/tests/dashboard/service_test.go:85
2023-09-13T09:39:00.1852388Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:101
2023-09-13T09:39:00.1853381Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:154
2023-09-13T09:39:00.1854312Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:76
2023-09-13T09:39:00.1854590Z             	Error:      	Not equal: 
2023-09-13T09:39:00.1854935Z             	            	expected: 401
2023-09-13T09:39:00.1855274Z             	            	actual  : 508
2023-09-13T09:39:00.1855733Z             	Test:       	TestDashboardTestSuite/TestDashboardProxy
2023-09-13T09:39:00.1855974Z         service_test.go:85: 
2023-09-13T09:39:00.1856585Z             	Error Trace:	/home/runner/work/pd/pd/tests/dashboard/service_test.go:85
2023-09-13T09:39:00.1857536Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:101
2023-09-13T09:39:00.1858527Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:154
2023-09-13T09:39:00.1859467Z             	            				/home/runner/work/pd/pd/tests/dashboard/service_test.go:76
2023-09-13T09:39:00.1859733Z             	Error:      	Not equal: 
2023-09-13T09:39:00.1860075Z             	            	expected: 401
2023-09-13T09:39:00.1860414Z             	            	actual  : 508
2023-09-13T09:39:00.1860987Z             	Test:       	TestDashboardTestSuite/TestDashboardProxy
2023-09-13T09:39:00.1861248Z [2023/09/13 09:38:59.479 +00:00] [INFO] [allocator_manager.go:316] ["exit allocator loop"] []
2023-09-13T09:39:00.1861342Z FAIL
```

https://github.com/tikv/pd/actions/runs/6170653223/job/16747325679?pr=7087

--- a/server/config/config.go
+++ b/server/config/config.go
@@ -505,7 +505,7 @@ type PDServerConfig struct {
 	// RuntimeServices is the running extension services.
 	RuntimeServices typeutil.StringSlice `toml:"runtime-services" json:"runtime-services"`
 	// MetricStorage is the cluster metric storage.
-	// Currently we use prometheus as metric storage, we may use PD/TiKV as metric storage later.
+	// Currently, we use prometheus as metric storage, we may use PD/TiKV as metric storage later.
 	MetricStorage string `toml:"metric-storage" json:"metric-storage"`
 	// There are some values supported: "auto", "none", or a specific address, default: "auto"
 	DashboardAddress string `toml:"dashboard-address" json:"dashboard-address"`

--- a/tests/dashboard/service_test.go
+++ b/tests/dashboard/service_test.go
@@ -86,7 +86,8 @@ func (suite *dashboardTestSuite) checkRespCode(url string, code int) {
 }
 
 func waitForConfigSync() {
-	time.Sleep(time.Second)
+	// Need to wait dashboard service start.
+	time.Sleep(3 * time.Second)
 }
 
 func (suite *dashboardTestSuite) checkServiceIsStarted(internalProxy bool, servers map[string]*tests.TestServer, leader *tests.TestServer) string {

